Dependencies.md

```diff
@@ -0,0 +1,58 @@
+
+The latter also serves as a time to open/close issues that are related to the updates. There might be a bug fix for an open issue - we should test it and close the issue. Or there might be new functionality that can be used - probably an issue should be open.
+
+### Exceptions:
+There are some dependencies that we really don't use all that much, intend on removing and as a general note don't need anything else from them. Given that we currently have no problems that updates will fix - we prefer to not update them as not to introduce bugs. Also, for some they bring additional dependencies that we do not want, which is just one more reason not to update them.
+
+List (as of March 2022):
+- github.com/DataDog/datadog-go  - newer versions have a lot more dependencies for functionality we don't need. Also in general a different library is probably going to be better in this case as it only supports UDP and no TCP.
+- github.com/andybalholm/cascadia - a dependency of `github.com/PuerkitoBio/goquery`
+- github.com/sirupsen/logrus - it's in maintenance mode and we want to remove it - also no update for a long time, but also no bugs.
+- github.com/spf13/afero - there are plans to be [replaced by io/fs](https://github.com/grafana/k6/issues/1079) and we don't need anything from it. We have already worked around some bugs so updating might break something
+- github.com/spf13/cobra - none of the newer features are particularly needed, but adds a bunch of new dependencies. Might want to replace it with https://github.com/muesli/coral
+- github.com/spf13/pflag - similar to above
+- gopkg.in/guregu/null.v3 - no new interesting features and we probably want to drop it in favor of [croconf](https://github.com/grafana/croconf)
+- gopkg.in/yaml.v3 - no new features wanted - actually used directly in only one place to output yaml to stdout.
+- google.golang.org/genproto, google.golang.org/protobuf, github.com/golang/protobuf - get update by `google.golang.org/grpc` when needed
+
+
+### How to do it:
+
+Currently, I will recommend using [modtools](https://github.com/dop251/modtools).
+
+Running `modtools check` will give you a list of packages that aren't frozen (the ones above in the exceptions).  Also, important that this lists both direct and indirect, so some care need to be taken. Alternatively just running `go get <dependency>` for each direct one also will tell you if there were update.
+
+Then go at look at the changelog between the version we are at and the latest version.
+ 
+You can use the command `modtools check` provides you to update it. Run test and if relevant check that bugs are fixed or any other verification that is appropriate.
+
+Commit dependencies one by one with a message like `Update <dependency> from vX.Y.Z to vX.Y.Z` and a relevant changelog for k6. Sometimes that means "nothing of relevant for k6", sometimes it means a list of bugfixes or new features.
+
+It's preferable to make multiple PRs - in most cases you can split them in three:
```

Comment:
```suggestion
It's preferable to make multiple PRs or split across different commits - in most cases you can split them in three:
```

A single PR with three different commits should be fine too, or not?

Reply:
I am really against updating multiple *direct* dependencies in the same commit unless necessary. 

It's both harder to write commit messages for those and it's more likely something will be missed. 

Also arguably making smaller PRs will be a lot easier to review ;)